_posts/2020-04-28-mocking.adoc

```diff
@@ -0,0 +1,365 @@
+---
+layout: post
+title: 'Mocking CDI beans in Quarkus'
+date: 2020-04-28
+tags: testing
+synopsis: Quarkus 1.4 introduces some interesting new capabilities for mocking CDI beans.
+author: geoand
+---
+
+Testing Quarkus applications has been an important part of the Quarkus Developer Joy, which is why
+`@QuarkusTest` for testing JVM applications and `@NativeTest` for black-box testing of the native images have been
+part of Quarkus since the first release.
+A recurring request however amongst our community members has been to have Quarkus allow them to selectively
+mock certain CDI beans for specific tests. This post will introduce the new mocking capabilities
+that `1.4.0.Final` brings which aim to address those concerns, while also providing a glimpse of additional
+improvements in this are that will be part of `1.5.0.Final`.
+
+== Old approach
+
+Let us assume that a Quarkus application contains the following (purely contrived) bean:
+
+[source,java]
+----
+@ApplicationScoped
+public class OrderService {
+
+	private final InvoiceService invoiceService;
+	private final InvoiceNotificationService invoiceNotificationService;
+
+    public OrderService(InvoiceService invoiceService, InvoiceNotificationService invoiceNotificationService) {
+    	this.invoiceService = invoiceService;
+    	this.invoiceNotificationService = invoiceNotificationService;
+    }
```

Comment:
Minor but indentation is a bit weird for the properties and this block of code.

Reply:
Fixed